centreon/src/Core/Dashboard/Domain/Model/DashboardRights.php

```diff
@@ -47,6 +49,36 @@ public function canAccess(): bool
         return $this->hasViewerRole();
     }
 
+    public function canDelete(DashboardSharingRoles $roles): bool
+    {
+        return DashboardSharingRole::Editor === $roles->getTheMostPermissiveRole();
+    }
+
+    public function canUpdate(DashboardSharingRoles $roles): bool
+    {
+        return DashboardSharingRole::Editor === $roles->getTheMostPermissiveRole();
+    }
+
+    public function canCreateShare(DashboardSharingRoles $roles): bool
+    {
+        return DashboardSharingRole::Editor === $roles->getTheMostPermissiveRole();
+    }
+
+    public function canDeleteShare(DashboardSharingRoles $roles): bool
+    {
+        return DashboardSharingRole::Editor === $roles->getTheMostPermissiveRole();
+    }
+
+    public function canUpdateShare(DashboardSharingRoles $roles): bool
+    {
+        return DashboardSharingRole::Editor === $roles->getTheMostPermissiveRole();
+    }
+
+    public function canAccessShare(DashboardSharingRoles $roles): bool
+    {
+        return null !== $roles->getTheMostPermissiveRole();
+    }
```

Comment:
These methods could have been moved to the DashboardSharingRoles object

Reply:
Yes, I could have put this un the `SharingRoles` classe, but I wanted to centralized all the permissions check inside the same object and not split this everywhere.